File: 520_Detect_Capital.java
class Solution {
    public boolean detectCapitalUse(String word) {
        int caps = 0;
        int n = word.length();
        for(int i = 0;i < n; i++){
            if(Character.isUpperCase(word.charAt(i))) caps++;
        }
        if(caps == 0) return true;
        else if(caps == n) return true;
        else if(caps == 1 && Character.isUpperCase(word.charAt(0))) return true;
        else return false;
    }
}

File: 520_Detect_Capital.rs
impl Solution {
    pub fn detect_capital_use(word: String) -> bool {
        let caps = word.chars().filter(|c| c.is_uppercase()).count();
        let n = word.len();
        
        if caps == 0 {
            true
        } else if caps == n {
            true
        } else if caps == 1 && word.chars().next().unwrap().is_uppercase() {
            true
        } else {
            false
        }
    }
}
